flushing_batch.go

```diff
@@ -62,6 +62,7 @@ func (b *singleIndexFlushingBatch) Flush() error {
 	if err != nil {
 		return err
 	}
-	b.batch.Reset()
+	// WARN: start a new flush and dont Reset it because it may result in a datarace
```

Comment:
wording nit: `// Create a new batch, because batches cannot be modified once they have been applied.`

Reply:
@ethantkoenig It seems bleve search has fixed the problem. And I will close this PR since this is unnecessary now.